### ISSUE TITLE
Monitor crashes when a result gets purged before downloading

### ISSUE THREAD

**Issue by donkirkby** [cfe-lab/MiCall#343]
Occasionally, we're very unlucky, and the results get purged between the time the sample is reprocessed and the time we download the results.

Options when a sample's results have been purged:
* Just reprocess the whole folder.
* Check whether any other results have been purged, and reprocess any that were purged.
* Download results as you go. Problem with sort order within a file? Better to use a subfolder for each file?

Also consider rerunning samples that have been stopped. This would be helpful for samples that get stuck by issue cfe-lab/Kive#566. Would it be too surprising for users? This would only be for samples that get stopped, not samples that fail.

Here's the plan:
* [x] Poll the status of *all* runs, including completed runs, while waiting for a batch to finish.
* [x] If a run gets purged or stopped, rerun it.
* [x] Catch exceptions during download, and mark the folder as failed.
* [x] Create runs in a batch, so that Kive won't purge them until the batch is finished. (Requires cfe-lab/Kive#434.)

Here's an example that we think happened because of a purged result:

~~~
2016-07-30 15:22:59[INFO]kive_loader.download_results(): downloading 
results for '/path/to/run'
2016-07-30 15:24:41[WARNING]kiveapi._validate_response(): Error response 
404 for https://kive-host/api/datasets/None/download/: 
{"detail":"Not found."}
2016-07-30 15:24:41[ERROR]MISEQ_MONITOR.main(): Fatal error.
Traceback (most recent call last):
   File "./MISEQ_MONITOR.py", line 60, in main
     delay = loader.poll()
   File 
"/path/to/micall/monitor/kive_loader.py", line 
87, in poll
     self.check_run_status()
   File 
"/path/to/micall/monitor/kive_loader.py", line 
188, in check_run_status
     self.download_results(folder, runs)
   File 
"/path/to/micall/monitor/kive_loader.py", line 
533, in download_results
     download_results(runs, results_folder, run_folder)
   File 
"/path/to/micall/monitor/kive_download.py", 
line 92, in download_results
     for j, line in enumerate(dataset.readlines()):
   File "build/bdist.linux-x86_64/egg/kiveapi/dataset.py", line 67, in 
readlines
     for line in self._request_download().iter_lines():
   File "build/bdist.linux-x86_64/egg/kiveapi/dataset.py", line 59, in 
_request_download
     stream=True)
   File "build/bdist.linux-x86_64/egg/kiveapi/kiveapi.py", line 132, in get
     is_json=is_json)
   File "build/bdist.linux-x86_64/egg/kiveapi/kiveapi.py", line 101, in 
_validate_response
     raise KiveServerException('Resource not found!')
KiveServerException: Resource not found!
~~~

**Comment by donkirkby** [comment]
First option is implemented in commit d6cf0fc95cf19f31431b5b7441964a99eb34973c. Not tested yet, and we're considering the other options before going with that one.

**Comment by donkirkby** [comment]
It happened again, and it looks like it was definitely caused by a purge. Here's the start of the response to the `run_outputs` request:

```
{
    "id": 107842,
    "output_summary": [
        {
            "is_ok": true,
            "name": "remap_counts",
            "url": null,
            "is_invalid": false,
            "display": "1: remap_counts",
            "redaction_plan": null,
            "step_name": "Run outputs",
            "date": "removed",
            "filename": null,
            "type": "dataset",
            "id": null,
            "size": "removed"
        },
        ...
```

We can detect the problem by checking for a null `id`.
